src/webapi/date.rs

```diff
@@ -1,25 +1,1166 @@
+                    @{self.as_ref()}.setUTCSeconds(@{seconds});
+                };
+            }
+        }
+    }
+
+    // DEPRECATED
+    // fn set_year() -> i32 { }
+
+    /// The to_date_string() method returns the date portion of a Date object in human readable form
+    /// in American English.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toDateString)
+    #[inline]
+    fn to_date_string( &self ) -> String {
+        js! ( return @{self.as_ref()}.toDateString(); ).try_into().unwrap()
+    }
+
+    // DEPRECATED
+    // fn to_gmt_string( &self ) -> i32 { }
+
+    /// The to_iso_string() method returns a string in simplified extended ISO format (ISO 8601),
+    /// which is always 24 or 27 characters long (YYYY-MM-DDTHH:mm:ss.sssZ or
+    /// ±YYYYYY-MM-DDTHH:mm:ss.sssZ, respectively). The timezone is always zero UTC offset, as
+    /// denoted by the suffix "Z".
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString)
+    #[inline]
+    fn to_iso_string( &self ) -> String {
+        js! ( return @{self.as_ref()}.toISOString(); ).try_into().unwrap()
+    }
+
+    /// The to_json() method returns a string representation of the Date object.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toJSON)
+    #[inline]
+    fn to_json( &self ) -> String {
+        js! ( return @{self.as_ref()}.toJSON(); ).try_into().unwrap()
+    }
+
+    /// The to_locale_date_string() method returns a string with a language sensitive representation
+    /// of the date portion of this date. The new locales and options arguments let applications
+    /// specify the language whose formatting conventions should be used and allow to customize the
+    /// behavior of the function. In older implementations, which ignore the locales and options
+    /// arguments, the locale used and the form of the string returned are entirely implementation
+    /// dependent.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString)
+    #[inline]
+    fn to_locale_date_string( &self, _locales: Option< &str >, _options: Option< &HashMap< &str, &str > >) -> String {
+        match _locales {
+            Some(_locales) => match _options {
+                Some(_options) => return js! (
+                    var _options = @{_options};
+
+                    return @{self.as_ref()}.toLocaleDateString(@{_locales}, _options);
+                 ).try_into().unwrap(),
+
+                 None => return js! (
+                     return @{self.as_ref()}.toLocaleDateString(@{_locales});
+                 ).try_into().unwrap()
+            },
+
+            None => return js! ( return @{self.as_ref()}.toLocaleDateString(); ).try_into().unwrap()
+        }
+
+    }
```

Comment:
Get rid for now. (:

Reply:
I can see ```toLocaleDateString```, ```toLocaleString```, and ```toLocaleTimeString``` being useful without specifying the locale and options explicitly. If someone is OK with the defaults, then these calls may still be beneficial for simple date formatting/presentation

Reply:
Hmm... okay, that sounds fine to me, especially since the original JS API can be called that way. We can add an ~~overload~~ another method later which will take the arguments if we wish so.